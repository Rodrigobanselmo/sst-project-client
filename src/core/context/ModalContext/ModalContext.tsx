/* eslint-disable no-unused-vars */
import React, { createContext, FC, useContext } from 'react';

import { useControlModal } from '../../hooks/useControlModal/useControlModal';

interface ModalContextData {
  isOpen: (name: string) => boolean;
  registerModal: (name: string) => {
    open: boolean;
    onClose: () => string[] | undefined;
  };
  openByName: (name: string) => void;
  closeByName: (name?: string | undefined) => string[] | undefined;
  onCloseAll: () => void;
}

const ModalContext = createContext({} as ModalContextData);

export const ModalProvider: FC = ({ children }) => {
  const { isOpen, registerModal, openByName, closeByName, onCloseAll } =
    useControlModal();
  return (
    <ModalContext.Provider
      value={{ isOpen, registerModal, openByName, closeByName, onCloseAll }}
    >
      {children}
    </ModalContext.Provider>
  );
};

export const useModal = (): ModalContextData => useContext(ModalContext);
