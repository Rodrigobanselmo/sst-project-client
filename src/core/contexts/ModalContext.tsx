/* eslint-disable no-unused-vars */
import React, { createContext, FC, useContext } from 'react';

import { useControlModal } from '../hooks/useControlModal';

interface ModalContextData {
  isOpen: (name: string) => boolean;
  registerModal: (name: string) => {
    open: boolean;
    onClose: () => string[] | undefined;
  };
  onOpenModal: (name: string) => void;
  onCloseModal: (name?: string | undefined) => string[] | undefined;
  onCloseAllModals: () => void;
}

const ModalContext = createContext({} as ModalContextData);

export const ModalProvider: FC = ({ children }) => {
  const { isOpen, registerModal, onOpenModal, onCloseModal, onCloseAllModals } =
    useControlModal();
  return (
    <ModalContext.Provider
      value={{
        isOpen,
        registerModal,
        onOpenModal,
        onCloseModal,
        onCloseAllModals,
      }}
    >
      {children}
    </ModalContext.Provider>
  );
};

export const useModal = (): ModalContextData => useContext(ModalContext);
