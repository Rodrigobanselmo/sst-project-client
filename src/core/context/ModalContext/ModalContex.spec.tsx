/* eslint-disable @typescript-eslint/no-explicit-any */
import React, { FC } from 'react';

import { render } from '@testing-library/react';

import * as ModalContext from './ModalContext';

const ChildrenSetup: FC = () => {
  const modal = ModalContext.useModal();
  return <p>{JSON.stringify(Object.keys(modal))}</p>;
};

describe('ModalContext', () => {
  it('i should return context value', () => {
    const { asFragment } = render(
      <ModalContext.ModalProvider>
        <ChildrenSetup />
      </ModalContext.ModalProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('i should return same context value for multiple children', () => {
    const { asFragment } = render(
      <ModalContext.ModalProvider>
        <ChildrenSetup />
        <ChildrenSetup />
      </ModalContext.ModalProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
