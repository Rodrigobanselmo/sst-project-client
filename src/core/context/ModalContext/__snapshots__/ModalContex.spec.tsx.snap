// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContext i should return context value 1`] = `
<DocumentFragment>
  <p>
    ["isOpen","registerModal","openByName","closeByName","onCloseAll"]
  </p>
</DocumentFragment>
`;

exports[`ModalContext i should return same context value for multiple children 1`] = `
<DocumentFragment>
  <p>
    ["isOpen","registerModal","openByName","closeByName","onCloseAll"]
  </p>
  <p>
    ["isOpen","registerModal","openByName","closeByName","onCloseAll"]
  </p>
</DocumentFragment>
`;
